### PR TITLE
Reformat markdown files

### LINKED ISSUE #985: Markdown files should be formatted consistently throughout this repository
The markdown files within this repository are not formatted, or at least not formatted consistently, making them hard to read. According to [the syntax guide][0], there are several improvements we can make:

* Line breaks around headers and code blocks (as well as in between paragraphs, if any exist)
* Consistent list item markers (`*` **or** `-`, not both)
* Remove numbered headers, and ordered, numbered lists where appropriate

[0]: https://daringfireball.net/projects/markdown/

### PR DESCRIPTION
**Pull request by sudoforge**
This patch reformats the markdown files in this repository with the overall goal of making them cleaner, more readable, and simpler to maintain.

closes #985

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -13,9 +13,11 @@ about: Create a report to help us improve
 [troubleshooting]: https://github.com/autozimu/LanguageClient-neovim/blob/next/INSTALL.md#6-troubleshooting
```
Comment (autozimu) [inline .github/ISSUE_TEMPLATE/bug_report.md]: The linked section needs to be updated with this PR.

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -1,29 +1,35 @@
...
 ```vim
 let g:LanguageClient_devel = 1 " Use rust debug build
 let g:LanguageClient_loggingLevel = 'DEBUG' " Use highest logging level
 ```
 
-1. Make necessary changes.
-1. Build. `make` to build, format and run [clippy], or `make build` to run build only.
-1. Verify changes.
-1. Run tests. (See below section)
+- Make necessary changes.
```
Comment (autozimu) [inline .github/CONTRIBUTING.md:16]: It's easier to reference the item when they're numbered. But I'm fine unifying list as well.

**Review by autozimu — COMMENTED** [review]
Thanks for contributing. 2 small comments.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 1 above)
Comment (sudoforge) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Great catch! I'll run a check for other references to any of the markdown files.

**Comment by autozimu** [comment]
Thanks!

**Pull request merged**